wp/plugin/wp.ml

```diff
@@ -207,6 +207,36 @@ let func_name_map = Cmd.parameter Typ.(list ~sep:';' (pair ~sep:',' string strin
            "<reg1_orig>,<reg1_mod>;<reg2_orig>,<reg2_mod>"). By default, WP
            assumes subroutines have the same names between the two binaries.|}
 
+let fun_specs = Cmd.parameter Typ.(list string) "fun-specs"
+    ~doc:{|List of built-in function summaries to be used at a function call
+           site in order of precedence. A target function will be mapped to a
+           function spec if it fulfills the spec's requirements. All function
+           specs set the target function as called and update the stack
+           pointer. The default spec will set the function as called and update
+           the stack pointer. Available built-in specs:
+
+           `verifier-error': Trips calls to __assert_fail.
+
+           `verifier-assume': Adds an assumption to the precondition based on
+           the argument to the function call.
+
+           `verifier-nondet': Chaoses the output to the function call
```

Comment:
It feels like these correspond to specific functions.

Reply:
I'll add this too.

Reply:
👍